src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -40,6 +56,17 @@ public override async Task<LSP.CompletionItem[]> HandleRequestAsync(LSP.Completi
                 return Array.Empty<LSP.CompletionItem>();
             }
 
+            // C# and VB share the same LSP language server, and thus share the same default trigger characters.
+            // We need to ensure the trigger character is valid in the document's language. For example, the '{'
+            // character, while a trigger character in VB, is not a trigger character in C#.
+            var triggerCharacter = char.Parse(request.Context.TriggerCharacter);
+            if (request.Context.TriggerKind != LSP.CompletionTriggerKind.Invoked && !char.IsLetterOrDigit(triggerCharacter) &&
```

Comment:
let's check specifically that it's a triggercharacter invocation instead of !Invoked.  We wouldn't want to check trigger chars if the LSP adds a deletion trigger kind for example